mods/cnc/rules/ai.yaml

```diff
@@ -289,3 +184,140 @@ Player:
 		Bots: hal9001
 	HarvesterBotModule:
 		RequiresCondition: enable-cabal-ai || enable-watson-ai || enable-hal9001-ai
+	BaseBuilderBotModule@cabal:
+		RequiresCondition: enable-cabal-ai
+		BuildingQueues: Building.Nod, Building.GDI
+		DefenseQueues: Defence.Nod, Defence.GDI
+		MinimumExcessPower: 30
+		MaximumExcessPower: 150
+		ExcessPowerIncrement: 30
+		ExcessPowerIncreaseThreshold: 5
+		BuildingFractions:
+			proc: 20%
+			pyle: 5%
+			hand: 5%
+			hq: 4%
+			weap: 9%
+			afld: 9%
+			gtwr: 5%
+			gun: 5%
+			atwr: 9%
+			obli: 7%
+			sam: 7%
+			eye: 1%
+			tmpl: 1%
+			silo: 0%
+			fix: 1%
+			hpad: 2%
+		BuildingCommonNames:
+			ConstructionYard: fact
+			Refinery: proc
+			Power: nuke,nuk2
+			Barracks: pyle,hand
+			VehiclesFactory: weap,afld
+			Production: pyle,hand,weap,afld,hpad
+			Silo: silo
+		BuildingLimits:
+			proc: 4
+			pyle: 3
+			hand: 3
+			hq: 1
+			weap: 3
+			afld: 3
+			hpad: 0
+			eye: 1
+			tmpl: 1
+			fix: 1
+			silo: 1
+	BaseBuilderBotModule@watson:
+		RequiresCondition: enable-watson-ai
+		BuildingQueues: Building.Nod, Building.GDI
+		DefenseQueues: Defence.Nod, Defence.GDI
+		MinimumExcessPower: 30
+		MaximumExcessPower: 150
+		ExcessPowerIncrement: 30
+		ExcessPowerIncreaseThreshold: 4
+		BuildingFractions:
+			proc: 17%
+			pyle: 2%
+			hand: 2%
```

Comment:
The automatic % &rarr; float conversions isn't used anywhere else, IIRC.  This would be a good opportunity to switch these over to a different syntax.

Reply:
Do you mean something like integer percentages, or something different entirely?